### PR TITLE
Allison's Admin page functionality

### PR DESCRIPTION
**Pull request by Dpalazzari**
In this PR:

Worked with @kheppenstall 

- Add Admin Component
- Add Cohort Component
- Add Admin link to Sidebar component
- Add Student Component
- Add studentList component.
- Add Group Component
- Add GroupList component
- Add SubmitGroup Component
- Add delete group component

TL;DR Allison's Admin page is fully functional with the ability to create and delete groups and it works with the back-end API.

Needs review by both of you, probably. @lucyconklin @wlffann 

FYI @case-eee 

[#142810047]

### CODE REVIEW

**Comment by Dpalazzari** [comment]
Check it out @lucyconklin @allisonreusinger @wlffann 

screen shot 2017-04-11 at 3 50 46 pm

**Comment by Dpalazzari** [comment]
Working on fixing Travis.CI

**Comment by Dpalazzari** [comment]
Travis.CI is failing on some npm run build component. All tests pass.

File: src/cohort/Cohort.js [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    this.toggleStudent = this.toggleStudent.bind(this);
+    this.submit = this.submit.bind(this);
+    this.delete = this.delete.bind(this);
+    this.addToList = this.addToList.bind(this);
+    this.post = this.post.bind(this);
+  }
+
+  post(){
+    console.log(JSON.stringify(this.state.groups))
+    fetch(`https://johariwindowapi.herokuapp.com/api/v1/assignments`, {
```
Comment (lucyconklin) [inline src/cohort/Cohort.js:19]: We should do a sweep of the app tomorrow and look for console.logs. I will add a chore to the tracker.

File: src/cohort/Cohort.js [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+  delete(group){
+    var groups = this.state.groups
+    this.addToList(group);
+    let id = group[0].id
+    for(var i = groups.length - 1; i > -1; i--){
+      for(var j = groups[i].length - 1; j > -1; j--){
+        if(groups[i][j].id === id){
+          groups.splice(i,1)
+        }
```
Comment (wlffann) [inline src/cohort/Cohort.js:38]: Phew! This is fun. Nested for loops!

File: src/cohort/Cohort.js [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+    this.retrieveStudents();
+  }
+
+  retrieveStudents(){
+    var that = this;
+    fetch('https://johariwindowapi.herokuapp.com/api/v1/cohorts/' + that.props.cohortID + '/users')
+      .then(result => result.json())
+      .then(data => {
+        that.setState({ students: data })
+        return true
```
Comment (wlffann) [inline src/cohort/Cohort.js:62]: We've got Axios in the next PR which could make this a little cleaner. You'll get JSON back as a default, I believe.

File: src/sidebar/Sidebar.js [hunk 4]
```diff
@@ -13,6 +13,7 @@ class Sidebar extends Component {
         <div className='sidebar-links'>
           <Link to='/'>Assignments</Link>
           <Link to='/mywindow'>My Window</Link>
+          <Link to='/admin'>Admin</Link>
```
Comment (wlffann) [inline src/sidebar/Sidebar.js:16]: 👍

**Review by wlffann — COMMENTED** [review]
I'm good with merging this. Thanks for putting this together @kheppenstall @Dpalazzari. Looks like it was complicated!

**Pull request merged**